lib/rubocop/cop/lint/struct_new_override.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks unexpected overrides of the `Struct` original methods
+      # via `Struct.new`.
+      #
+      # @example
+      #   # bad
+      #   Bad = Struct.new(:members, :clone, :count)
+      #   b = Bad.new([], true, 1)
+      #   b.members #=> [] (overriding `Struct#members`)
+      #   b.clone #=> true (overriding `Object#clone`)
+      #   b.count #=> 1 (overriding `Enumerable#count`)
+      #
+      #   # good
+      #   Good = Struct.new(:id, :name)
+      #   g = Good.new(1, "foo")
+      #   g.members #=> [:id, :name]
+      #   g.clone #=> #<struct Good id=1, name="foo">
+      #   g.count #=> 2
+      #
+      class StructNewOverride < Cop
```

Comment:
And then we can name this `StructBuiltInMethodOverride` which is more descriptive IMO.

Reply:
Thank you. I also think that `StructBuiltInMethodOverride` is easier to understand than `StructNewOverride`. 👍 
I'll fix it.

Reply:
I assume that this cop detects some overrides by arguments of `Struct.new`, not normal overrides as follows:

```ruby
Foo = Struct.new(:name) do
  def to_s
    name.to_s
  end
end
```

I'm a bit concerned that `StructBuiltInMethodOverride` may be considered as **any** overrides of the built-in methods. 🤔

Reply:
For example, `StructBuiltInMethodOverrideByConstructor` may be more descriptive but too long. 😓 
I may think too much... Please give me any help! 🙏

Reply:
Naming is hard! :D Let's keep the shorter name and down the road we can maybe extend this cop to cover dangerous overrides by explicit definitions as well.

Reply:
Make sense. Thanks! 👍